search-commons/src/main/java/no/unit/nva/search2/ResourceParameter.java

```diff
@@ -0,0 +1,203 @@
+package no.unit.nva.search2;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+import no.unit.nva.search2.model.IParameterKey;
+import nva.commons.core.JacocoGenerated;
+import org.jetbrains.annotations.NotNull;
+
+import static java.util.Objects.nonNull;
+import static no.unit.nva.search2.constant.ErrorMessages.ERROR_MESSAGE_INVALID_VALUE;
+import static no.unit.nva.search2.constant.ErrorMessages.ERROR_MESSAGE_INVALID_DATE;
+import static no.unit.nva.search2.constant.ErrorMessages.ERROR_MESSAGE_INVALID_FIELD_VALUE;
+import static no.unit.nva.search2.constant.ErrorMessages.ERROR_MESSAGE_INVALID_NUMBER;
+import static no.unit.nva.search2.constant.Patterns.PATTERN_IS_DATE;
+import static no.unit.nva.search2.constant.Patterns.PATTERN_IS_NON_EMPTY;
+import static no.unit.nva.search2.constant.Patterns.PATTERN_IS_NUMBER;
+import static no.unit.nva.search2.constant.Patterns.PATTERN_IS_SHORT_DATE;
+import static no.unit.nva.search2.model.IParameterKey.Operator.EQUALS;
+import static no.unit.nva.search2.model.IParameterKey.Operator.GREATER_THAN;
+import static no.unit.nva.search2.model.IParameterKey.Operator.GREATER_THAN_OR_EQUAL_TO;
+import static no.unit.nva.search2.model.IParameterKey.Operator.LESS_THAN;
+import static no.unit.nva.search2.model.IParameterKey.Operator.NONE;
+import static no.unit.nva.search2.model.IParameterKey.ParamKind.CUSTOM;
+import static no.unit.nva.search2.model.IParameterKey.ParamKind.DATE;
+import static no.unit.nva.search2.model.IParameterKey.ParamKind.NUMBER;
+import static no.unit.nva.search2.model.IParameterKey.ParamKind.SHORT_DATE;
+import static no.unit.nva.search2.model.IParameterKey.ParamKind.STRING;
+
+public enum ResourceParameter implements IParameterKey {
```

Comment:
This class define all parameters -> transformations, errormessages, comparar, validation for key and valuess